neard/src/watchers/mod.rs

```diff
@@ -0,0 +1,60 @@
+pub mod dyn_config_watcher;
+pub mod log_config_watcher;
+
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+use std::io;
+use std::io::ErrorKind;
+use std::path::PathBuf;
+use tracing::{error, info};
+
+pub(crate) enum UpdateBehavior {
+    UpdateOrReset,
+    UpdateOnlyIfExists,
+}
+
+#[derive(thiserror::Error, Debug)]
+#[non_exhaustive]
+pub(crate) enum WatchConfigError {
+    #[error("Failed to reload the watcher config")]
+    Reload(String),
+    #[error("Failed to reload the logging config")]
+    Parse(#[source] serde_json::Error),
+    #[error("Can't open or read the logging config file")]
+    OpenAndRead(#[source] io::Error),
+}
+
+pub(crate) trait Watcher
```

Comment:
as this is a trait - let's add some comments on how it should be used, what does it do etc.